packages/ng-primitives/combobox/src/combobox-option/combobox-option.ts

```diff
@@ -0,0 +1,72 @@
+import { BooleanInput } from '@angular/cdk/coercion';
+import { booleanAttribute, computed, Directive, input, OnDestroy, OnInit } from '@angular/core';
+import { injectElementRef } from 'ng-primitives/internal';
+import { uniqueId } from 'ng-primitives/utils';
+import { injectComboboxState } from '../combobox/combobox-state';
+
+@Directive({
+  selector: '[ngpComboboxOption]',
+  exportAs: 'ngpComboboxOption',
+  host: {
+    role: 'option',
+    '[id]': 'id()',
+    '[attr.aria-selected]': 'selected() ? "true" : undefined',
+    '[attr.data-selected]': 'selected() ? "" : undefined',
+    '[attr.data-disabled]': 'disabled() ? "" : undefined',
+  },
+})
+export class NgpComboboxOption<T> implements OnInit, OnDestroy {
+  /** Access the combobox state. */
+  protected readonly state = injectComboboxState<T | T[]>();
+
+  /**
+   * The element reference of the option.
+   * @internal
+   */
+  readonly elementRef = injectElementRef();
+
+  /** The id of the option. */
+  readonly id = input<string>(uniqueId('ngp-combobox-option'));
+
+  /** The value of the option. */
+  readonly value = input<T>(undefined, {
+    alias: 'ngpComboboxOptionValue',
+  });
+
+  /** The disabled state of the option. */
+  readonly disabled = input<boolean, BooleanInput>(false, {
+    alias: 'ngpComboboxOptionDisabled',
+    transform: booleanAttribute,
+  });
+
+  /** Whether this option is selected. */
+  protected readonly selected = computed(() => {
+    const value = this.value();
+
+    if (!value) {
+      return false;
+    }
+
+    if (this.state().multiple()) {
+      return Array.isArray(value) && value.includes(this.value());
+    }
+
+    return value === this.value();
+  });
```

Comment:
This looks like a bug

L43: `const value = this.value();`
L51: `value.includes(this.value())`
L54: `return value === this.value();`

Reply:
Good spot 😊 thank you!

Reply:
How about something like "compareWith" in ng-select ? it's very important as signals force people to make deep copies and other things so we may need to check if option is selected using some object key e.g id.

Reply:
Absolutely 😊 it's on my todo list for tonight! Thanks!